### ISSUE TITLE
Docker network doesn't create any calico network

### ISSUE THREAD

**Issue by mcortinas** [projectcalico/calicoctl#1507]
I cannot create docker network , I'm running CentOs7:

```
# docker network create --driver=calico --ipam-driver=calico-ipam my-calico-net
Error response from daemon: failed to update store for object type *libnetwork.endpointCnt: client: etcd cluster is unavailable or misconfigured
```

Other thest in order to describe my environment
```
# curl http://IP_ETCD_SERVER:2379/version
{"etcdserver":"3.1.0","etcdcluster":"3.1.0"}

# ps -ef|grep docker|grep daemon
root      7846     1 99 10:22 ?        02:55:38 /usr/bin/docker daemon --cluster-store=etcd://IP_ETCD_SERVER:2379 --cluster-advertise=br25:2379 -H unix:///var/run/docker.sock --ip-forward=true --iptables=true --ip-masq=true --log-driver journald --storage-driver=overlay

# /usr/local/bin/calicoctl node checksystem
Checking kernel version...
                3.10.0-327.36.3.el7.x86_64                                      OK
Checking kernel modules...
                xt_set                                                  OK
                ip6_tables                                              OK
System meets minimum system requirements to run Calico!

# /usr/local/bin/calicoctl node run
Running command to load modules: modprobe -a xt_set ip6_tables
Enabling IPv4 forwarding
Enabling IPv6 forwarding
Increasing conntrack limit
Removing old calico-node container (if running).
Running the following command to start calico-node:

docker run --net=host --privileged --name=calico-node -d --restart=always -e ETCD_SCHEME= -e NODENAME=IP_CALICO_SERVER_NAME -e CALICO_NETWORKING_BACKEND=bird -e NO_DEFAULT_POOLS= -e CALICO_LIBNETWORK_ENABLED=true -e CALICO_LIBNETWORK_IFPREFIX=cali -e ETCD_ENDPOINTS=http://IP_ETCD_SERVER:2379 -e ETCD_AUTHORITY= -v /lib/modules:/lib/modules -v /var/log/calico:/var/log/calico -v /run/docker/plugins:/run/docker/plugins -v /var/run/docker.sock:/var/run/docker.sock -v /var/run/calico:/var/run/calico calico/node:v1.0.2

Image may take a short time to download if it is not available locally.
Container started, checking progress logs.
Waiting for etcd connection...
Using configured IPv4 address: IP_CALICO_SERVER
No IPv6 address configured
Using global AS number
Calico node name:  IP_CALICO_SERVER_NAME
CALICO_LIBNETWORK_ENABLED is true - start libnetwork service
Calico node started successfully

# docker version
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:11 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:11 2016
 OS/Arch:      linux/amd64

```

**Comment by mcortinas** [comment]
I added other commands suggested in slack channel
```
``` (1:1194)# docker exec calico-node /bin/sh -c 'wget -O - $ETCD_ENDPOINTS/version'
Connecting to 10.2.X.Y:2379 (10.2.X.Y:2379)
-                    100% |*******************************|    44   0:00:00 ETA```
....
(1:1195)# docker logs calico-node
time="2017-02-08T13:20:13Z" level=info msg="Creating Calico client" 
time="2017-02-08T13:20:13Z" level=info msg="Loading config from environment" 
time="2017-02-08T13:20:13Z" level=info msg="Ensuring datastore is initialized" 
time="2017-02-08T13:20:13Z" level=info msg="Ready flag is already set" 
time="2017-02-08T13:20:13Z" level=info msg="Datastore is initialized" 
time="2017-02-08T13:20:13Z" level=info msg="Ensuring a cluster guid is set" 
time="2017-02-08T13:20:13Z" level=info msg="Using previously configured ClusterGUID" 
Waiting for etcd connection...
Using configured IPv4 address: 10.2.X.Y
No IPv6 address configured
Using global AS number
Calico node name:  BAREMETAL_HOSTNAME
mv: can't remove '/etc/service/available/felix/log': Directory not empty
mv: can't remove '/etc/service/available/felix': Directory not empty
mv: can't remove '/etc/service/available/bird/log': Directory not empty
mv: can't remove '/etc/service/available/bird': Directory not empty
mv: can't remove '/etc/service/available/bird6/log': Directory not empty
mv: can't remove '/etc/service/available/bird6': Directory not empty
mv: can't remove '/etc/service/available/confd/log': Directory not empty
mv: can't remove '/etc/service/available/confd': Directory not empty
CALICO_LIBNETWORK_ENABLED is true - start libnetwork service
mv: can't remove '/etc/service/available/libnetwork/log': Directory not empty
mv: can't remove '/etc/service/available/libnetwork': Directory not empty
Calico node started successfully
```

**Comment by mcortinas** [comment]
Finally i enabled trace loglevel in docker daemon and i can see this error stack trace.
```
error level=error msg="discovery error: Unexpected watch error"
```
I solved downgrade ETCD to 2.3.7 and its works fine!
